modules/packages/UnrolledLinkedList.chpl

```diff
@@ -0,0 +1,1245 @@
+        _enter();
+        result = _insertGenericKnownSize(idx, lst, size);
+        _leave();
+      }
+
+      return result;
+    }
+
+    /*
+      Remove the first `count` elements from this unrolledLinkedList with values equal to
+      `x`, shifting all elements following the removed item left.
+
+      If the count of elements to remove is less than or equal to zero, then
+      all elements from this unrolledLinkedList equal to the value of `x` will be removed.
+
+      .. warning::
+
+        Removing elements from this unrolledLinkedList may invalidate existing references
+        to the elements contained in this unrolledLinkedList.
+
+      :arg x: The value of the element to remove.
+      :type x: `eltType`
+
+      :arg count: The number of elements to remove.
+      :type count: `int`
+
+      :return: The number of elements removed.
+      :rtype: `int`
+    */
+    proc ref remove(x: eltType, count:int=1): int {
+      var result = 0;
+
+      on this {
+        _enter(); defer _leave();
+
+        var removed = 0;
+
+        // For each node
+        var cur = _head;
+        while cur != nil && (removed < count || count == 0) {
+          // The following is similar to the original List's remove
+
+          // Run through the leading non-x prefix
+          var dst_i = 0;
+          while dst_i < cur!.size {
+            if cur!.data[dst_i] == x {
+              removed += 1;
+              break;
+            }
+            dst_i += 1;
+          }
+          
+          // No hit in this node, continue
+          if dst_i == cur!.size {
+            cur = cur!.next;
+            continue;
+          }
+
+          // Now dst_i should be the index of the first `x`
+          var src_i = dst_i + 1;
+          while src_i < cur!.size {
+            ref src = cur!.data[src_i];
+            if (count == 0 || removed < count) && src == x {
+              src_i += 1;
+              removed += 1;
+              continue;
+            }
+            ref dst = cur!.data[dst_i];
+            if src_i != dst_i then dst = src;
+            src_i += 1;
+            dst_i += 1;
+          }
+
+          cur!.size = dst_i;
+
+          cur = cur!.next;
+        }
+
+        // Maybe merge nodes
+        cur = _head;
+        while cur != nil {
+          try! {
+            var result = _merge(cur!);
+            if result == false then cur = cur!.next;
+          }
+        }
+
+        result = removed;
+        _size -= removed;
+      }
+
+      return result;
+    }
+
+    pragma "no doc"
+    proc ref _popAtIndex(idx: int): eltType {
+      _size -= 1;
+
+      var cur = _head;
+      var pos = 0;
+      while cur != nil {
+        if idx < pos + cur!.size {
+          var result = cur!.data[idx-pos];
+
+          // Shift elements
+          cur!.size -= 1;
+          for i in idx-pos..<cur!.size {
+            cur!.data[i] = cur!.data[i+1];
+          }
+
+        try! {
+          _merge(cur: unmanaged _linkedNode(eltType));
```

Comment:
If `cur` is the last node, and it contained only the one element being removed, this will leave `tail` empty, no?